src/main/java/duke/task/DukeList.java

```diff
@@ -0,0 +1,233 @@
+
+    /**
+     * Adds an Event task to the list.
+     *
+     * @param text Body of the task to be added.
+     * @return The message associated to the task being displayed.
+     */
+    public String addEvent(String text) {
+        String[] strings = text.split(" /at ", 2);
+
+        String limit = strings.length == 1 ? "" : strings[1];
+
+        Event input = new Event(strings[0].trim(), limit);
+
+        tasks.add(input);
+
+        return displayTask(input);
+    }
+
+    /**
+     * Lists out the current tasks in the list.
+     *
+     * @return The list of tasks in the list.
+     */
+    public String list() {
+        String str = "Here are the tasks in your list:\n";
+        for (int i = 0; i < tasks.size(); i++) {
+            str += (i + 1) + "." + tasks.get(i).toString() + "\n";
+        }
+
+        return str;
+    }
+
+    /**
+     * Marks a task in the list as done.
+     *
+     * @param item Index of the task marked as done.
+     * @return The message associated to the task being done.
+     */
+    public String done(int item) {
+        Task task = tasks.get(item - 1);
+        task.setDone();
+
+        String response = "Nice! I've marked this task as done:\n";
+        return response + task.toString();
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param item Index of the task to be deleted.
+     * @return The message associated to the task being deleted.
+     */
+    public String delete(int item) {
+        Task task = tasks.get(item - 1);
+        tasks.remove(item - 1);
+
+        String response = "Noted. I've removed this task:\n";
+        String taskCount = "\nNow you have " + tasks.size() + " tasks in the list.";
+        return response + task.toString() + taskCount;
+    }
+
+    /**
+     * Finds all tasks in list with the input keyword.
+     *
+     * @param input User input keyword.
+     * @return The response to the number of tasks found.
+     */
+    public String find(String input) {
+        String trimmedInput = input.trim();
+        String str = "";
+        String msg;
+
+        int count = 0;
+
+        boolean doesContain;
+
+        for (int i = 0; i < tasks.size(); i++) {
+            msg = tasks.get(i).toString();
+            doesContain = msg.contains(trimmedInput);
+
+            if (doesContain) {
+                str += (i + 1 + "." + msg + "\n");
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            return "Sorry no tasks found that match your input.";
+        } else {
+            return str + "Are these the tasks you're looking for?";
+        }
+    }
+
+
+    /**
+     * Returns the string form of the DukeList object.
+     *
+     * @return The string form of the DukeList.
+     */
+    @Override
+    public String toString() {
+        String str = "";
+
+        for (int i = 0; i < tasks.size(); i++) {
+            str += tasks.get(i).saveData() + "\n";
+        }
+
+        return str;
+    }
+
+
```

Comment:
May want to check line spacing between methods? Here there's two line spaces but for some methods above there's only one in between methods